### PR TITLE
Added pause menu

### PR DESCRIPTION
**Pull request by Gabriel-Sous-a**
Added pause menu, didn't add logic to it so the buttons won't work this will be done in a later date.

Screenshot 2023-09-16 013501

[Ticket](https://github.com/FlamingoFiestaStudio/OlympusGoneWild/issues/12)

### CODE REVIEW

File: pauseMenu.tscn
Comment (201flaviosilva) [inline pauseMenu.tscn:1]: Why this scene?

File: UI/pause_menu.tscn [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+anchor_left = 0.5
+anchor_top = 0.5
+anchor_right = 0.5
+anchor_bottom = 0.5
+offset_left = -127.0
+offset_top = -160.0
+offset_right = 123.0
+offset_bottom = 158.0
+grow_horizontal = 2
+grow_vertical = 2
```
Comment (201flaviosilva) [inline UI/pause_menu.tscn:27]: Why isn't this using the entire screen?

File: UI/pause_menu.tscn [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+[node name="VBoxContainer" type="VBoxContainer" parent="Panel"]
+layout_mode = 1
+anchors_preset = 15
+anchor_right = 1.0
+anchor_bottom = 1.0
+grow_horizontal = 2
+grow_vertical = 2
+
+[node name="Label" type="Label" parent="Panel/VBoxContainer"]
```
Comment (201flaviosilva) [inline UI/pause_menu.tscn:37]: It would have changed to `TitleLabel`

**Review by 201flaviosilva — APPROVED** [review]

File: pauseMenu.tscn
Comment (Gabriel-Sous-a) [reply pauseMenu.tscn:1]: that was just to test, I'm going to remove it in this [ticket](https://github.com/FlamingoFiestaStudio/OlympusGoneWild/issues/31), but I can remove it now if you think its better

File: UI/pause_menu.tscn (hunk 1 above)
Comment (Gabriel-Sous-a) [reply UI/pause_menu.tscn:27]: the `control` above is using the entire screen and the `panel` is just centered

File: UI/pause_menu.tscn (hunk 2 above)
Comment (Gabriel-Sous-a) [reply UI/pause_menu.tscn:37]: as a type?

File: pauseMenu.tscn
Comment (201flaviosilva) [reply pauseMenu.tscn:1]: Yap, I think it's better, just to ensure this PR goes perfect 💯

File: UI/pause_menu.tscn (hunk 2 above)
Comment (201flaviosilva) [reply UI/pause_menu.tscn:37]: Nop, as a name

Comment (201flaviosilva) [reply UI/pause_menu.tscn:37]:
```suggestion
[node name="TitleLabel" type="Label" parent="Panel/VBoxContainer"]
```

File: UI/pause_menu.tscn (hunk 1 above)
Comment (201flaviosilva) [reply UI/pause_menu.tscn:27]: Well, since we don't have designs yet... 🤷‍♂️

**Review by Adan0k — APPROVED** [review]

File: pauseMenu.tscn
Comment (Gabriel-Sous-a) [reply pauseMenu.tscn:1]: done

**Pull request merged**